integration-tests/mock-components/updatable-promise-page.js

```diff
@@ -0,0 +1,27 @@
+const { registerHtml, useObservable } = require('../../src/tram-one')
+const home = require('./home')
+const updatedPromise = require('./updated-promise')
+
+const html = registerHtml({
+	home,
+	'updated-effect': updatedPromise
+})
+
+module.exports = () => {
+	const [count, setCount] = useObservable(5)
+	const [showCount, setShowCount] = useObservable(true)
+
+	const increment = () => setCount(count + 1)
+
+	const removeCount = () => {
+		setShowCount(false)
+	}
+
+	return html`
+		<home>
+			<button data-testid="increment-count" onclick=${increment}>Increment</button>
+			<button data-testid="remove-count" onclick=${removeCount}>Remove Count</button>
+			${showCount ? html`<updated-effect initialCount=${count} />` : 'Counter Removed'}
```

Comment:
the whole purpose of this component is to be able to trigger the cleanup of a component... in reality, this should be refactored so that all kinds of cleanup effects can be validated... something that should happen when the test code is cleaned up (in another PR... probably).